### PR TITLE
CartRepositoryImpl.GetShoppingCartsByIds was optimized

### PR DESCRIPTION
**Pull request by trueboroda**
this return  empty collection when get empty request param

### CODE REVIEW

File: VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs [hunk 1]
```diff
@@ -246,55 +246,62 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
...
         {
-            var result = CustomerOrders.Where(x => ids.Contains(x.Id)).ToArray();
-            var orderDiscounts = Discounts.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
-            var orderTaxDetails = TaxDetails.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
-
-            if (responseGroup.HasFlag(CustomerOrderResponseGroup.WithAddresses))
+            if (ids.IsNullOrEmpty())
             {
-                var addresses = Addresses.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
+                return new CustomerOrderEntity[0];
```
Comment (artem-dudarev) [inline VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: Single return

File: VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs [hunk 2]
```diff
@@ -246,55 +246,62 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
...
             {
-                var inPayments = InPayments.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
-                var paymentsIds = inPayments.Select(x => x.Id).ToArray();
-                var paymentDiscounts = Discounts.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-                var paymentTaxDetails = TaxDetails.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-                var paymentAddresses = Addresses.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-                var transactions = Transactions.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-            }
+                var result = CustomerOrders.Where(x => ids.Contains(x.Id)).ToArray();
+                var orderDiscounts = Discounts.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
```
Comment (artem-dudarev) [inline VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: Get actual ids from `result`

File: VirtoCommerce.OrderModule.Data/Services/CustomerOrderServiceImpl.cs [hunk 3]
```diff
@@ -59,7 +59,8 @@ public virtual void SaveChanges(CustomerOrder[] orders)
             using (var repository = RepositoryFactory())
             using (var changeTracker = GetChangeTracker(repository))
             {
-                var dataExistOrders = repository.GetCustomerOrdersByIds(orders.Where(x => !x.IsTransient()).Select(x => x.Id).ToArray(), CustomerOrderResponseGroup.Full);
+                var orderIds = orders.Where(x => !x.IsTransient()).Select(x => x.Id).ToArray();
```
Comment (yecli) [inline VirtoCommerce.OrderModule.Data/Services/CustomerOrderServiceImpl.cs]: I think we do not need to make that change as it is the only one in the file and it gives nothing to the task.

**Comment by tatarincev** [comment]
Stale PR

**Pull request closed without merging**